Update README to provide Docker instructions for Mosquitto broker (#1090)
aws/aws-iot-device-sdk-embedded-C: README.md

1. Go to the `build/bin` directory and run any demo executables from there.

> ### Alternative option of Docker containers for running demos locally.

Reviewer: I think the header level got flipped with the next sections.

Author: Fixed

1. Go to the `build/bin` directory and run any demo executables from there.

### Alternative option of Docker containers for running demos locally.

Install Docker:

```shell

curl -fsSL https://get.docker.com -o get-docker.sh

sh get-docker.sh

```

#### Installing Mosquitto to run MQTT demos locally

The following instructions have been tested on an Ubuntu 18.04 environment with Docker and OpenSSL installed.

1.  Download the official Docker image for Mosquitto.

```shell

docker pull eclipse-mosquitto:latest

```
2.  `BROKER_ENDPOINT` defined in `demos/mqtt/mqtt_demo_basic_tls/demo_config.h` can now be set to `localhost`.

3. For TLS communication with Mosquitto broker, server and CA credentials need to be created. Use OpenSSL commands to generate the credentials for the Mosquitto server.

Generate CA key and certificate. Provide the Subject field information as appropriate.
```shell
openssl req -x509 -nodes -sha256 -days 365 -newkey rsa:2048 -keyout ca.key -out ca.crt
```

> Generate server key and certificate by getting it signed by the CA cert.

Reviewer 2: Wording:
```suggestion
Generate server key and certificate and sign with the CA cert.
```
